### PR TITLE
[ASL-4413] HBA upload functionality in PPL transfer journeys

### PR DESCRIPTION
**Pull request by edemirbag**
- Provide a flexible mechanism for key-based lookup with fallback options in support of content updates related to ASL-4413
- Include tests to validate that the snippet component can correctly handle fallbacks when attempting to retrieve templates

### CODE REVIEW

File: src/snippet/index.jsx [hunk 1]
```diff
@@ -4,13 +4,27 @@ import { connect } from 'react-redux';
...
+
 export const Snippet = ({ content, children, optional, fallback, ...props }) => {
-    const str = get(content, children, get(content, fallback));
+    const str = getTemplate(content, children, fallback);
     if (str === undefined && optional) {
         return null;
+
     }
     if (str === undefined) {
-        throw new Error(`Failed to lookup content snippet: ${children}`);
```
Comment (farrukhishaq) [inline src/snippet/index.jsx:13]: is this intentional?

File: src/snippet/index.spec.jsx [hunk 2]
```diff
@@ -42,4 +42,16 @@ two`
     expect(wrapper.html()).toEqual(paragraphs);
   });
```
Comment (farrukhishaq) [inline src/snippet/index.spec.jsx:44]: this test added is not checking code changes. 

`return Failed to lookup content snippet: ${children}`

File: src/snippet/index.jsx [hunk 3]
```diff
@@ -4,13 +4,27 @@ import { connect } from 'react-redux';
...
+            return template;
+        }
+    }
+
+    return undefined;
+}
+
 export const Snippet = ({ content, children, optional, fallback, ...props }) => {
-    const str = get(content, children, get(content, fallback));
+    const str = getTemplate(content, children, fallback);
```
Comment (farrukhishaq) [inline src/snippet/index.jsx:21]: 👍🏼

**Review by farrukhishaq — COMMENTED** [review]
remove console log and clearify on content changes look for comments.

File: src/snippet/index.jsx (hunk 1 above)
Comment (edemirbag) [reply src/snippet/index.jsx:13]: Good spot. I'll undo this particular commit.

File: src/snippet/index.spec.jsx (hunk 2 above)
Comment (edemirbag) [reply src/snippet/index.spec.jsx:44]: as above, will undo that line - cheers

**Review by farrukhishaq — APPROVED** [review]

**Pull request merged**